### PR TITLE
updating parambulator to latest deps

### PR DESCRIPTION
**Pull request by RGCsAGupta**
- [x] replace `build.sh` & `test.sh` scripts with npm scripts
- [x] replace `travis` with `github` actions
- [x] replace `phantomjs` with `karma chromeheadless` test
- [x] replace `jsamine-node` with `jasmine`
- [x] bump `gex` from 0.2.2 to 1.0.0
- [x] bump `jsonic` from 0.2.2 to 0.3.1
- [x] bump `lodash` from 4.5.0 to 4.17.15
- [x] remove `chai`
- [x] bump `docco` from 0.7.0 to 0.8.0
- [x] bump `jshint` from 2.8.0 to 2.11.1
- [x] bump `uglify-js` from 2.4.23 to 3.10.0

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    branches: [ master , release/*]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [10.x, 12.x, 13.x, 14.x]
```
Comment (CExAdamJ) [inline .github/workflows/node.js.yml]: Let's remove 13.x IMO. Non-prod version for 6.0 we should be at 12 or hopefully 14 before release. This should save a little bit of github actions time.

File: .github/workflows/node.js.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Node.js CI
+
+on:
+  push:
+    branches: [ master, release/*, chore/* ]
```
Comment (CExAdamJ) [inline .github/workflows/node.js.yml]: Not sure our the objective here, but if its to run on push to PR branch it should have `feature/*` and `push/*`. I would say this is unnecessary though. I think we can remove on push altogether and use branch protection instead.

File: .github/workflows/node.js.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  pull_request:
+    branches: [ master , release/*]
```
Comment (CExAdamJ) [inline .github/workflows/node.js.yml]: Let's add `develop` as well and fix the conventions in this repo.
```suggestion
    branches: [ master , develop , release/*]
```

File: .github/workflows/node.js.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [10.x, 12.x, 13.x, 14.x]
+
+    steps:
+    - uses: actions/checkout@v2
```
Comment (CExAdamJ) [inline .github/workflows/node.js.yml]: If we pin these versions to an exact version dependabot can suggest and test updates for us.

File: .github/workflows/npm-audit.yml [hunk 5]
```diff
@@ -0,0 +1,24 @@
+name: npm audit
+
+on:
+  pull_request:
+  push:
```
Comment (CExAdamJ) [inline .github/workflows/npm-audit.yml]: Same comments here.

File: .github/workflows/npm-audit.yml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: install dependencies
+        run: npm ci
+      - uses: oke-py/npm-audit-action@v1.5.0
+        with:
+          audit_level: moderate
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          # issue_assignees: oke-py // not needed
+          issue_labels: vulnerability,test
```
Comment (CExAdamJ) [inline .github/workflows/npm-audit.yml]: I believe the convention we used for labels in other repos is `security` and I dont think we need `test` here.

**Review by CExAdamJ — COMMENTED** [review]
Nice work

Some minor comments

File: .github/workflows/npm-audit.yml (hunk 6 above)
Comment (RGCsAGupta) [inline .github/workflows/npm-audit.yml]:
```suggestion
          issue_labels: vulnerability, security
```

File: .github/workflows/node.js.yml [hunk 7]
```diff
@@ -0,0 +1,29 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Node.js CI
+
+on:
+  push:
```
Comment (RGCsAGupta) [inline .github/workflows/node.js.yml]:
```suggestion
```

File: .github/workflows/node.js.yml [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [10.x, 12.x, 13.x, 14.x]
+
+    steps:
+    - uses: actions/checkout@v2
```
Comment (RGCsAGupta) [inline .github/workflows/node.js.yml]:
```suggestion
    - uses: actions/checkout@vv2.3.1
```

File: .github/workflows/npm-audit.yml (hunk 6 above)
Comment (RGCsAGupta) [reply .github/workflows/npm-audit.yml]: updated

File: .github/workflows/node.js.yml (hunk 4 above)
Comment (RGCsAGupta) [reply .github/workflows/node.js.yml]: updated

File: test/parambulator.spec.js [hunk 9]
```diff
@@ -1392,3 +1384,9 @@ describe('parambulator', function() {
 
 })
 
+
+// describe(' test suite', ()=>{
+// 	it('random test', ()=>{
+// 		expect('akash').toEqual('akash');
+// 	})
+// })
```
Comment (RGCsAGupta) [inline test/parambulator.spec.js]:
```suggestion

```

File: .github/workflows/node.js.yml [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+    branches: [ master , develop , release/*]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [10.x, 12.x, 13.x, 14.x]
```
Comment (CExAGupta) [inline .github/workflows/node.js.yml:18]:
```suggestion
        node-version: [10.x, 12.x, 14.x]
```

File: .github/workflows/node.js.yml (hunk 1 above)
Comment (RGCsAGupta) [reply .github/workflows/node.js.yml]: removed 13.x

Comment (RGCsAGupta) [reply .github/workflows/node.js.yml]:
```suggestion
        node-version: [10.x, 12.x, 14.x]
```

File: .github/workflows/npm-audit.yml (hunk 5 above)
Comment (RGCsAGupta) [reply .github/workflows/npm-audit.yml]:
```suggestion
```

File: .github/workflows/node.js.yml [hunk 11]
```diff
@@ -0,0 +1,28 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Node.js CI
+
+on:
+    branches: [ master, release/*, chore/* ]
+  pull_request:
+    branches: [ master , develop , release/*]
```
Comment (RGCsAGupta) [inline .github/workflows/node.js.yml]:
```suggestion
on:
  pull_request:
    branches:
      - master
      - develop
      - 'release/*'
```

File: .github/workflows/npm-audit.yml [hunk 12]
```diff
@@ -0,0 +1,23 @@
+name: npm audit
+
+on:
+  pull_request:
+    branches:
+      - master
+      - 'releases/*'
```
Comment (RGCsAGupta) [inline .github/workflows/npm-audit.yml]:
```suggestion
on:
  pull_request:
    branches:
      - master
      - develop
      - 'release/*'
```

File: .github/workflows/node.js.yml [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [10.x, 12.x, 14.x]
+
+    steps:
+    - uses: actions/checkout@vv2.3.1
```
Comment (RGCsAGupta) [inline .github/workflows/node.js.yml]:
```suggestion
    - uses: actions/checkout@v2.3.1
```

File: .github/workflows/node.js.yml (hunk 2 above)
Comment (RGCsAGupta) [reply .github/workflows/node.js.yml]: updated

**Review by CExAdamJ — APPROVED** [review]
LGTM

**Pull request merged**